File: src/main/java/com/modak/notification/entities/NotificationEvent.java
package com.modak.notification.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
public class NotificationEvent {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    private Long id;


    @ManyToOne
    @JoinColumn(name = "notification_type_id")
    private NotificationType type;
    private String description;
    @Column(name = "event_time")
    private LocalDateTime eventTime;

}

File: src/main/java/com/modak/notification/infraestructure/notificationEvent/entities/NotificationEvent.kt
package com.modak.notification.infraestructure.notificationEvent.entities

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime


@Table("notification_event")
data class NotificationEvent (
    @Id val id: Long?=0,
    val notificationTypeId: Long,
    val sentAt: LocalDateTime
)
